### PR TITLE
Centos minimal 8

### PR DESCRIPTION
**Pull request by Happy2C0de**
Add support for Centos8.

### CODE REVIEW

**Comment by valerytschopp** [comment]
@Happy2C0de unfortunately I can not merge this in the SWITCHengines branch

The SWITCHengines branch is NOT on upstream master, but on tag 2.33.0 (commit 3a7d4528a06b553170e968e6fb8c187433548e15), but your branch is on master... So if I merge this it can create havoc.

Can you rebase your branch on the SWITCH switch-ch/diskimage-builder SWITCHengines branch?

```
git remote add switch https://github.com/switch-ch/diskimage-builder.git
git fetch --all
git checkout centos-minimal-8 
git status
git rebase switch/SWITCHengines
git status
git pull
git status
git diff switch/SWITCHengines
```

**Comment by valerytschopp** [comment]
@Happy2C0de a better branch name: `centos8`

**Comment by valerytschopp** [comment]
And can you add the `rm /etc/motd.d/cockpit` somewhere?

**Comment by Happy2C0de** [comment]
I rebased the branch and added a comment about the WebConsole. Furthermore, a note how one can remove the notification is in the CentOS 8 README.

**Comment by Happy2C0de** [comment]
A renaming of the branch closes the PR and I must create a new one. Does this make sense? I would keep it. Because the current name is not completely off the track.

**Comment by valerytschopp** [comment]
Nice, with the rebase, only 16 files have changed, compared to 122 before ;)

And it is ok to keep this branch name.

But personally, I would prefer to `rm /etc/motd.d/cockpit`. Experienced users know about Cockpit already, so they can enable it if they want. But unexperienced users will enable, set a bad user password, open the firewall, and will get hacked...

**Comment by valerytschopp** [comment]
Did you try to build a image with the rebased branch? Does it still work?

File: diskimage_builder/elements/centos8/post-install.d/23-cloud-init-default-user
Comment (valerytschopp) [inline diskimage_builder/elements/centos8/post-install.d/23-cloud-init-default-user]: Here the sed match is very broad. 

You should match `name: cloud-user` and replace by `name: $DIB_CLOUD_INIT_DEFAULT_USER` e.g.:

`sed -i -e "s/name: cloud-user/name: $DIB_CLOUD_INIT_DEFAULT_USER/g" /etc/cloud/cloud.cfg`

**Review by valerytschopp — CHANGES_REQUESTED** [review]

**Review by valerytschopp — APPROVED** [review]
Thanks

**Comment by Happy2C0de** [comment]
Branch was tested. Current branch works as expected. No ssh notification anymore and centos name works. 
Basic testing as mentioned here, https://wiki.smap.switch.ch/topic/switchcloud/images:release_process#test_scripts , was conducted. (plus a bit more)

Ready to merge I guess. Don't see any obvious errors. Images are published in staging (s1 and s2)
I will push it to production once it is merged.

Thanks for reviewing! :)

**Comment by valerytschopp** [comment]
Thanks Michael, excellent work!

**Pull request merged**
